From 52b4b81ad3b48d5e392dccf9fbbb97f52b1dcdc3 Mon Sep 17 00:00:00 2001
From: Arie Brown <abrown@habana.ai>
Date: Thu, 27 Aug 2020 17:27:55 +0300
Subject: [PATCH] [SW-19146] goya: dsv: create dsv fit

as part of creating goya boot and linux fit, the dsv fit should
also be created.

Change-Id: I3ff05e7b38f7ee197dbe2214b5cd5b83cdcd9c63
Signed-off-by: Arie Brown <abrown@habana.ai>
---

diff --git a/board/habanalabs/goya/post-image-fit.sh b/board/habanalabs/goya/post-image-fit.sh
new file mode 100755
index 0000000..cd1f942
--- /dev/null
+++ b/board/habanalabs/goya/post-image-fit.sh
@@ -0,0 +1,18 @@
+
+DSV_BIN=${COMMON_BASE}/dsv.bin
+DSV_ITS_FILE=dsv.its
+DSV_ITS_TEMPLATE=${COMMON_BASE}/${DSV_ITS_FILE}
+DSV_ITB_FILE=dsv-fit.itb
+
+function generate_device_fit
+{
+	echo "generate_dev_fit"
+	pushd ${BINARIES_DIR}
+
+		cp ${DSV_ITS_TEMPLATE} ${DSV_ITS_FILE}
+		${HOST_DIR}/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f ${DSV_ITS_FILE} ${DSV_ITB_FILE}
+		# LKD like this to be 4B aligned in size
+		dd status=none conv=sync bs=4 if=${UBOOT_ITB_FILE} > ${HL_DEV}-boot-fit.itb
+
+	popd 
+}
diff --git a/board/habanalabs/goya2/post-image-fit.sh b/board/habanalabs/goya2/post-image-fit.sh
new file mode 100755
index 0000000..cd1f942
--- /dev/null
+++ b/board/habanalabs/goya2/post-image-fit.sh
@@ -0,0 +1,18 @@
+
+DSV_BIN=${COMMON_BASE}/dsv.bin
+DSV_ITS_FILE=dsv.its
+DSV_ITS_TEMPLATE=${COMMON_BASE}/${DSV_ITS_FILE}
+DSV_ITB_FILE=dsv-fit.itb
+
+function generate_device_fit
+{
+	echo "generate_dev_fit"
+	pushd ${BINARIES_DIR}
+
+		cp ${DSV_ITS_TEMPLATE} ${DSV_ITS_FILE}
+		${HOST_DIR}/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f ${DSV_ITS_FILE} ${DSV_ITB_FILE}
+		# LKD like this to be 4B aligned in size
+		dd status=none conv=sync bs=4 if=${UBOOT_ITB_FILE} > ${HL_DEV}-boot-fit.itb
+
+	popd 
+}
diff --git a/package/dsv/dsv.mk b/package/dsv/dsv.mk
index be5b31a..b54aac8 100644
--- a/package/dsv/dsv.mk
+++ b/package/dsv/dsv.mk
@@ -8,4 +8,9 @@
 	$(TARGET_MAKE_ENV) $(MAKE) CROSS_COMPILE="$(TARGET_CROSS)" -C $(@D) all
 endef
 
+DSV_INSTALL_IMAGES = YES
+define DSV_INSTALL_IMAGES_CMDS
+        $(INSTALL) -D -m 0644 $(@D)/dsv.bin $(BINARIES_DIR)/
+endef
+ 
 $(eval $(generic-package))
